README.md
<h2 align='center'> Vendere Payments Flutter </h2>

<h3>Description</h3>
This project has digital payments functionalities for POS smart terminals or smartphones.

## Getting Started

This project is a starting point for a Flutter application.

A few resources to get you started if this is your first Flutter project:

- [Lab: Write your first Flutter app](https://docs.flutter.dev/get-started/codelab)
- [Cookbook: Useful Flutter samples](https://docs.flutter.dev/cookbook)

For help getting started with Flutter development, view the
[online documentation](https://docs.flutter.dev/), which offers tutorials,
samples, guidance on mobile development, and a full API reference.

[Versionamento de apps](https://stackoverflow.com/questions/53570575/flutter-upgrade-the-version-code-for-play-store)

[Package-name-fluter](https://stackoverflow.com/questions/51534616/how-to-change-package-name-in-flutter)

<img src="https://lh3.googleusercontent.com/pw/ADCreHfa5ePQngBosf_DbxVbN3PKuuX0dNK0NYfenTOtAKxzmYMajY1ea0vWM1J-5y0KI_AJPMA65Eu4RiWDisr_-sQDCbbBMtxplPH3OusumxY4WBiFtZw7M_k2iwh71x6eW5Xzpz1mCXCzrh40UEPSDR5Jm1OcHOUeY6TUTYY22JvMVWyH6hqd2dy-bhoRf2HN9vy1IH8jQ3d9hiay10JJVj4SuBFLm6bixpMajYXp_Zl8x4yStfk_VFxop6Ah5hUq2tMW_i6lPZxteGIysW5HUhUzV2preQdYMRSDJbIAb6kwPqkuGYhzD_LNESwmQ7b6QX_XduNMzDaXaEENmBXkzkzBbvZs8pKrhiiwGkCyTrq1awHBEoIkIR2UXLo7XAcJuxN4Mgy-PW1FCAVMiODBxwny8-3NUbrudAW9uSXLssc4PfH-KzC18j_6zf6kH7y_6_aQq3-ucIUaKy_Nra9eku3cYFJkQRHwGxcd_O8-vojBdktkq5PsaaRzWBRVXPIYG3QJkAD_uwONyyz2VwJKzU4mCNUvFyD_fTzId304jCGakDu_jHBT6I2NG_uUc2NbfodV5tC5EOainKH6rjR3t-ZBEGv7U26U8l5lzeV-t4gB0AC_GhiP8LuzcG52c7vM7OdsVwSuZoiZaOb_Gmi-j9LeVtAwQ5llI1inO7-_6oqQGtQpIC-q7S15Y1o3ZZzvvRFv4P2A461Wbki2jYEje8TAlTQdhTHRCdSmWPiTPhDUF5Gcc-QHn71pqtFdFBmycyYigBz-FQk4bkuQ2EjNxnxoyKIsO3iqlZDd3Nt2ynFvRzxc-N7EuZq4vWnUUqWIRzLOjVSAOWrfpLEyC6xv6x0mCC5ZUISt8WpjD-cH4WUn8JVaDT_Ev3-PikOd4fUcx8bvTq2d1P6mXO1WWb3dTw=w1311-h931-s-no?authuser=0" />

CHANGELOG.md
Change Log All notable changes to this project will be documented in this file. This change log follows the conventions of keepachangelog.com.

[0.0.1] 23-10-10 Added #0001 | Primeiras funcionalidades do projeto.

lib/main_screen.dart
import 'package:flutter/material.dart';

class MainScreen extends StatefulWidget {
  const MainScreen({super.key});

  @override
  State<MainScreen> createState() => _MainScreenState();
}

class _MainScreenState extends State<MainScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.spaceAround,
        children: [
          Padding(
            padding: const EdgeInsets.only(top: 30),
            child: Text("Selecione o pagamento:", style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold, color: Colors.blue[800])),
          ),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              Column(
                children: [
                  GestureDetector(
                    onTap: () {
                      print("pressionado credito");
                    },
                    child: Icon(
                      Icons.credit_card,
                      size: 130,
                      color: Colors.blue[800],
                    ),
                  ),
                  Text(
                    "CRÉDITO",
                    style: TextStyle(fontSize: 30, fontWeight: FontWeight.bold, color: Colors.blue[800]),
                  ),
                ],
              ),
              Column(
                children: [
                  GestureDetector(
                    onTap: () {
                      print("pressionado débito");
                    },
                    child: Icon(
                      Icons.credit_score,
                      color: Colors.blue[800],
                      size: 130,
                    ),
                  ),
                  Text(
                    "DÉBITO",
                    style: TextStyle(fontSize: 30, fontWeight: FontWeight.bold, color: Colors.blue[800]),
                  ),
                ],
              )
            ],
          ),
          // padding: const EdgeInsets.only(top: 30),
          const Padding(padding: EdgeInsets.only(top: 1)),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            crossAxisAlignment: CrossAxisAlignment.end,
            children: [
              Column(
                children: [
                  GestureDetector(
                    onTap: () {
                      print("pressionado pix");
                    },
                    child: Icon(
                      Icons.pix,
                      color: Colors.blue[800],
                      size: 130,
                    ),
                  ),
                  Text(
                    "PIX",
                    style: TextStyle(fontSize: 30, fontWeight: FontWeight.bold, color: Colors.blue[800]),
                  ),
                ],
              ),
              Column(
                children: [
                  GestureDetector(
                    onTap: () {
                      print("pressionado split");
                    },
                    child: Icon(
                      Icons.call_split,
                      color: Colors.blue[800],
                      size: 130,
                    ),
                  ),
                  Text(
                    "SPLIT",
                    style: TextStyle(fontSize: 30, fontWeight: FontWeight.bold, color: Colors.blue[800]),
                  ),
                ],
              ),
            ],
          ),
          const Padding(padding: EdgeInsets.only(top: 20)),
        ],
      ),
    );
  }
}


lib/payment_calculator.dart
import 'package:flutter/material.dart';

class PaymentCalculator extends StatefulWidget {
  const PaymentCalculator({super.key});

  @override
  State<PaymentCalculator> createState() => _PaymentCalculatorState();
}

class _PaymentCalculatorState extends State<PaymentCalculator> {
  final TextEditingController _paymentValue = TextEditingController();
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          // const TextField(decoration: InputDecoration(border: InputBorder(borderSide: BorderSide(color: Colors.red)))),
          TextField(
            controller: _paymentValue,
            keyboardType: TextInputType.number,
            style: const TextStyle(fontSize: 44),
            decoration: const InputDecoration(
              labelText: "R\$ 80,00",
              // prefixIconColor: Colors.blue[700],
              // suffixIconColor: Colors.blue[700],
              // iconColor: Colors.blue[700],
              alignLabelWithHint: true,
              floatingLabelAlignment: FloatingLabelAlignment.center,
              // contentPadding: EdgeInsets.only(top: 20, bottom: 20, left: 22, right: 22),
            ),
          ),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              IconButton(onPressed: () => print("var 1"), icon: Text("1", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 2"), icon: Text("2", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 3"), icon: Text("3", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
            ],
          ),
          const Padding(padding: EdgeInsets.all(10)),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              IconButton(onPressed: () => print("var 4"), icon: Text("4", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 5"), icon: Text("5", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 6"), icon: Text("6", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
            ],
          ),
          const Padding(padding: EdgeInsets.all(10)),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              IconButton(onPressed: () => print("var 7"), icon: Text("7", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 8"), icon: Text("8", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var 9"), icon: Text("9", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
            ],
          ),
          const Padding(padding: EdgeInsets.all(10)),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              IconButton(onPressed: () => print("var backspace"), icon: Icon(Icons.backspace, color: Colors.orange[800]), iconSize: 65),
              IconButton(onPressed: () => print("var 0"), icon: Text("0", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 60, color: Colors.blue[800])), iconSize: 65),
              IconButton(onPressed: () => print("var ="), icon: Icon(Icons.check, color: Colors.green[600]), iconSize: 65),
            ],
          ),
        ],
      ),
    );
  }
}


lib/home.dart
import 'dart:developer';

import 'package:flutter/material.dart';
import 'package:vendere_payments_flutter/payment_calculator.dart';

import 'main_screen.dart';

class Home extends StatefulWidget {
  const Home({super.key});

  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.blue[700],
        centerTitle: true,
        title: const Text(
          "Vendere",
          style: TextStyle(fontSize: 25),
        ),
        actions: [
          IconButton(
            icon: const Icon(
              Icons.notifications,
              size: 30,
            ),
            onPressed: () {
              print("ação account");
            },
          ),
          const Padding(padding: EdgeInsets.only(right: 15)),
        ],
      ),
      body: const MainScreen(),
      floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
      // floatingActionButton: FloatingActionButton(
      floatingActionButton: FloatingActionButton.extended(
        backgroundColor: Colors.blue[700],
        foregroundColor: Colors.white,
        // elevation: 10,
        icon: const Icon(
          Icons.calculate,
          size: 30,
        ),
        label: const Text("Calculadora", style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold)),
        // shape: BeveledRectangleBorder(borderRadius: BorderRadius.circular(4)),
        onPressed: () {
          log("Navigating to calculator");
          Navigator.push(context, MaterialPageRoute(builder: (context) => const PaymentCalculator()));
        },
        // mini: true,
        // child: const Icon(Icons.add),
        // onPressed: () {
        //   print("Pressionado");
        // },
      ),
      bottomNavigationBar: BottomAppBar(
        height: 70,
        // shape: const CircularNotchedRectangle(), //shape para botão circular
        child: Row(
          children: [
            const Padding(padding: EdgeInsets.only(left: 16)),
            IconButton(
              onPressed: () {},
              icon: Icon(Icons.note_add, size: 30, color: Colors.blue[800]),
            ),
            const Padding(padding: EdgeInsets.only(left: 10)),
            IconButton(
              onPressed: () {},
              icon: Icon(Icons.payment, size: 30, color: Colors.blue[800]),
            ),
            const Padding(padding: EdgeInsets.only(left: 165)),
            IconButton(
              onPressed: () {},
              icon: Icon(Icons.account_circle, size: 30, color: Colors.blue[800]),
            ),
            const Padding(padding: EdgeInsets.only(left: 5)),
            IconButton(
              onPressed: () {},
              icon: Icon(Icons.add_business, size: 30, color: Colors.blue[800]),
            ),
          ],
        ),
      ),
      drawer: Drawer(
        child: Padding(
          padding: const EdgeInsets.only(top: 20, left: 1),
          child: SafeArea(
              child: Column(
            children: [
              const ListTile(
                  // dense: true,
                  // title: Image.asset("images/ps-paystore.png", height: 40),
                  // onTap: () {
                  //   //action when this menu is pressed
                  // },
                  ),
              const Divider(color: Colors.grey),
              ListTile(
                dense: true,
                title: Text("DADOS DO TERMINAL", style: TextStyle(fontSize: 16, color: Colors.blue[800])),
                leading: Icon(Icons.person, color: Colors.blue[800]),
                onTap: () {
                  //action when this menu is pressed
                  // Navigator.push(context, MaterialPageRoute(builder: (context) => const TerminalData()));
                },
              ),
              ListTile(
                dense: true,
                title: Text("INICIALIZAR", style: TextStyle(fontSize: 16, color: Colors.blue[800])),
                leading: Icon(Icons.send, color: Colors.blue[800]),
                onTap: () {
                  const CircularProgressIndicator();
                  //action when this menu is pressed
                },
              ),
              ListTile(
                dense: true,
                title: Text("PAGAMENTOS", style: TextStyle(fontSize: 16, color: Colors.blue[800])),
                leading: Icon(Icons.payments, color: Colors.blue[800]),
                onTap: () {
                  // Navigator.push(context, MaterialPageRoute(builder: (context) => const Payments()));
                },
              ),
              ListTile(
                dense: true,
                title: Text("My Favourites", style: TextStyle(fontSize: 16, color: Colors.blue[800])),
                leading: Icon(Icons.monitor_heart, color: Colors.blue[800]),
                onTap: () {
                  //action when this menu is pressed
                },
              ),
              ExpansionTile(
                title: Text("CONFIGURAÇÕES", style: TextStyle(fontSize: 16, color: Colors.blue[800])),
                leading: Icon(Icons.settings, color: Colors.blue[800]), //add icon
                childrenPadding: const EdgeInsets.only(left: 60), //children padding
                children: [
                  ListTile(
                    title: const Text("Child Category 1", style: TextStyle(fontSize: 16)),
                    onTap: () {
                      //action on press
                    },
                  ),
                  ListTile(
                    title: const Text("Child Category 2", style: TextStyle(fontSize: 16)),
                    onTap: () {
                      //action on press
                    },
                  ),
                ],
              ),
              Padding(
                padding: const EdgeInsets.only(top: 20),
                child: Text(
                  "V1.0.0",
                  style: TextStyle(fontSize: 16, color: Colors.blue[800]),
                ),
              ),
            ],
          )),
        ),
      ),
    );
  }
}
